server/server.go

```diff
@@ -36,19 +52,101 @@ func upload(w http.ResponseWriter, r *http.Request) {
 	w.Write(responseS)
 }
 
-// Reads the compressed file and invokes function to decompress it
+// Builds and runs the assignment uploaded.
+func buildRun(w http.ResponseWriter, r *http.Request) {
+
+	fmt.Println("Build and Run Endpoint Hit")
+	w.Header().Set("Access-Control-Allow-Origin", "*")
+
+	var outputString string
+	var currDir string
+	var err error
+
+	outputString, currDir = navigateToWorkDir()
+	if outputString == "" {
+		outputString, err = runCommand(AssignmentData.CommandToCompile)
+
+		if err == nil {
+			finalCmd := AssignmentData.CommandToExecute
+			for _, value := range AssignmentData.CmdlineArgs {
+				finalCmd = fmt.Sprintf("%s %s", finalCmd, value)
+			}
+			fmt.Println(finalCmd)
+			outputString, err = runCommand(finalCmd)
+		}
+
+		err = os.Chdir(currDir)
+		if err != nil {
+			log.Fatalf("error while navigating to the current directory: %v", err)
+		}
+	}
+
+	response, err := json.Marshal(outputString)
+	if err != nil {
+		log.Println(err)
+	}
+	// Write the response to be sent to client.
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(http.StatusOK)
+	w.Write(response)
+}
+
+func navigateToWorkDir() (string, string) {
+	var responseS = ""
+	workDir := filepath.Join(AssignmentData.RootDir, AssignmentData.WorkDir)
+	currDir, _ := os.Getwd()
+	err := os.Chdir(filepath.Join(currDir, constants.AssignmentsDir, workDir))
```

Comment:
from this it seems like `RootDir` and `WorkDir` are relative pathnames. If that is the case, then this should be mentioned on the UI so that students don't input fully qualified path names.

Reply:
Okay